jscomp/others/belt_Array.ml

```diff
@@ -511,3 +596,13 @@ let unzip a =
     a2.!(i) <- v2
   done;
   (a1, a2)
+
+let joinWith a sep =
+  match length a with
+  | 0 -> ""
+  | l ->
+      let lastIndex = l - 1 in
+      let rec aux i res =
+        if i = lastIndex then res ^ a.!(i)
```

Comment:
could use some getUnsafe for perf

Reply:
I saw this on top of `Belt_Array.ml`:
```ocaml
external getUnsafe: 'a array -> int -> 'a = "%array_unsafe_get"
external (.!()): 'a array -> int -> 'a = "%array_unsafe_get"
```
and thought that `a.!(i)` and `getUnsafe(a, i)` was equivalent. 
Isn't it the case?